hack/dev-functions.sh

```diff
@@ -0,0 +1,95 @@
+	prevcontext=$(kubectl config current-context)
+
+	kind create cluster --name "$clustername"
+
+	kind get kubeconfig --name "$clustername" --internal >/tmp/"$clustername".yml
+	kubectl config use-context kind-argocd-control-plane
+	kubectl cp /tmp/"$clustername".yml argocd/"$argoserver":/tmp/"$clustername".yml
+	retry_argocd_exec "$argocdlogin && argocd cluster add kind-$clustername --kubeconfig /tmp/$clustername.yml --upsert" || echo "Success after retrying"
+	kubectl config use-context "$prevcontext"
+}
+
+function local_argocd_login() {
+
+	prevcontext=$(kubectl config current-context)
+	kubectl config use-context kind-argocd-control-plane >/dev/null
+
+	kubectl patch svc argocd-server -n argocd -p '{"spec": {"type": "NodePort"}}' >/dev/null
+	controlplane_container=$(docker ps -aqf "name=argocd-control-plane")
+	socat_container=$(docker ps -aqf "name=argocd-server-socat-proxy")
+	argoserver_nodeport=$(kubectl get service -n argocd argocd-server -o=jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+	argocdlogin="argocd login --insecure --username admin --password admin argocd-server.argocd.svc.cluster.local:443"
+
+	# Generate token for prc user
+	token=$(retry_argocd_exec "$argocdlogin >/dev/null && argocd account generate-token --account prc")
+	serverip=$(kubectl get service -n argocd argocd-server -o=jsonpath='{.spec.clusterIP}')
+
+	kubectl config use-context "$prevcontext" >/dev/null
+
+	docker stop "$socat_container" >/dev/null || err "No proxy container running"
+	docker rm "$socat_container" >/dev/null || err "No proxy container running"
+
+	# Kind isn't kind when it comes to exposing additional nodeports
+	# since they also need to be mapped to docker
+	# However, we can run socat to proxy a tcp socket without having to meddle with
+	# the kind controlplane container. See: https://github.com/kubernetes-sigs/kind/issues/99#issuecomment-456184883
+	docker run \
+		--name argocd-server-socat-proxy \
+		--detach \
+		--restart always \
+		--publish "$argoserver_nodeport":"$argoserver_nodeport" \
+		--link "$controlplane_container":target \
+		--network kind \
+		alpine/socat \
+		tcp-listen:"$argoserver_nodeport",fork,reuseaddr tcp-connect:target:"$argoserver_nodeport" >/dev/null
+
+	argocd login --insecure --username prc --password prc localhost:"$argoserver_nodeport" >/dev/null
+
+	{
+		echo "ARGOCD_TOKEN=$token"
+		echo "ARGOCD_USERNAME=prc"
+		echo "ARGOCD_PASSWORD=prc"
+		echo "ARGOCD_INCLUSTER_ADDRESS=$serverip"
+		echo "ARGOCD_LOCAL_ADDRESS=https://localhost:$argoserver_nodeport"
+	} >.env.local
+
+	# Create a secret storing the token and in-cluster ip
+	kubectl delete secret generic -n argocd prc-controller-secret || err "Secret not found. Creating.."
+	kubectl create secret generic -n argocd prc-controller-secret --from-literal="token=$token" --from-literal="serverip=$serverip" >/dev/null
+
+	echo "https://localhost:$argoserver_nodeport"
```

Comment:
```suggestion
	open "https://localhost:$argoserver_nodeport"
```
Could be worth opening it straight away?

Reply:
I need this to be the return value of function, so I have to `echo` it in here, but I can add `open` to the places where it is being called.